src/shims/foreign_items.rs

```diff
@@ -124,8 +130,26 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         // First: functions that diverge.
         match link_name {
-            "__rust_start_panic" | "panic_impl" => {
-                throw_unsup_format!("the evaluated program panicked");
+            // Note that this matches calls to the *foreign* item "__rust_start_panic" -
+            // that is, calls `extern "Rust" { fn __rust_start_panic(...) }`
+            // We forward this to the underlying *implementation* in "libpanic_unwind"
+            "__rust_start_panic" => {
+                let start_panic_instance = this.resolve_path(&["panic_unwind", "__rust_start_panic"])?;
```

Comment:
Shouldn't this be "panic_unwind" *or* "panic_abort", depending on compiler flags? AFAIK this is exactly where that switch is implemented.

Reply:
In fact, the user can also define arbitrary panic handlers of their own: https://github.com/rust-lang/rust/blob/9ebf47851a357faa4cd97f4b1dc7835f6376e639/src/test/compile-fail/auxiliary/panic-runtime-unwind2.rs

I believe what you want is https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyCtxt.html#method.injected_panic_runtime (and then call `crate_name`), but I'm not sure if that works for the `panic_unwind` and `panic_abort` crates.

Worst case you can do `tcx.crates()` and check `tcx.is_panic_runtime` on each of them

Reply:
@oli-obk: I've switched to using `injected_panic_runtime`.